File: old-projects/src/LevelManager.ts
import { LevelInfo } from "./LevelInfo";
import { Point } from "./Point";

export class LevelManager {
  static readonly levels: LevelInfo[] = [
    LevelManager.create("Bellingham", 80, 10, "bbbsssttt", [[100, 0], [100, 250], [350, 250], [200, 100], [350, 100], [600, 350], [100, 350]]),
    LevelManager.create("Everett", 65, 10, "bst", [[100, 0], [100, 300], [300, 300], [300, 150], [600, 150]]),
    LevelManager.create("Seattle", 60, 10, "bs", [[100, 0], [100, 100], [600, 200], [150, 350]]),
    LevelManager.create("Tacoma", 50, 10, "ttb", [[200, 0], [200, 300], [500, 300], [500, 100]]),
    LevelManager.create("Olympia", 11, 3, "bst", [[350, 0], [350, 350]]),
  ];

  private static create(name: string, monsterCount: number, allowEscape: number, sequence: string, points: number[][]): LevelInfo {
    const level = new LevelInfo();
    level.name = name;
    level.monsterCount = monsterCount;
    level.monstersAllowedEscape = allowEscape;
    level.monsterSequence = sequence;
    level.points = points.map(([x, y]) => new Point(x, y));
    return level;
  }
}

File: old-projects/vector-defence-silverlight-port/original-csharp-source/LevelManager.cs
using System;
using System.Collections.Generic;
using System.Windows;
using System.Windows.Resources;
using System.Xml;

namespace VectorDefenceSL
{
    public static class LevelManager
    {
        private static readonly List<LevelInfo> listOfLevels = new List<LevelInfo>();

        static LevelManager()
        {
            StreamResourceInfo sr = Application.GetResourceStream(new Uri("Levels.xml", UriKind.Relative));
            using (XmlReader reader = XmlReader.Create(sr.Stream))
            {
                while (reader.Read())
                {
                    if (reader.NodeType == XmlNodeType.Element && reader.Name == "level")
                    {
                        var level = LoadSingleLevel(reader);
                        listOfLevels.Add(level);
                    }
                }
            }
        }

        public static List<LevelInfo> Levels
        {
            get { return listOfLevels; }
        }

        private static LevelInfo LoadSingleLevel(XmlReader r)
        {
            var level = new LevelInfo { Name = r.GetAttribute("name") };
            while (r.Read())
            {
                if (r.NodeType == XmlNodeType.Element && r.Name == "p")
                {
                    double x = Convert.ToDouble(r.GetAttribute("x"));
                    double y = Convert.ToDouble(r.GetAttribute("y"));
                    level.Points.Add(new Point(x, y));
                }
                if (r.NodeType == XmlNodeType.Element && r.Name == "monsters")
                {
                    level.MonsterCount = Convert.ToInt32(r.GetAttribute("count"));
                    level.MonstersAllowedEscape = Convert.ToInt32(r.GetAttribute("allowEscape"));
                    level.MonsterSequence = r.ReadElementContentAsString();
                }
                if (r.NodeType == XmlNodeType.EndElement && r.Name == "level")
                {
                    break;
                }
            }
            return level;
        }
    }
}
